packages/admin/admin/src/dataGrid/CrudContextMenu.tsx

```diff
@@ -0,0 +1,193 @@
+
+        if (clipboard) {
+            let input;
+
+            try {
+                input = JSON.parse(clipboard);
+            } catch (e) {
+                errorDialog?.showError({
+                    userMessage: <FormattedMessage id="comet.common.clipboardInvalidFormat" defaultMessage="Clipboard contains an invalid format" />,
+                    error: e.toString(),
+                });
+                console.error("Bad clidpboard value, parsing JSON failed", e);
+            }
+
+            if (input) {
+                // TODO validate input?
+                try {
+                    await onPaste({
+                        input,
+                        client,
+                    });
+                    await client.refetchQueries({ include: refetchQueries });
+                } catch (e) {
+                    errorDialog?.showError({
+                        userMessage: (
+                            <FormattedMessage
+                                id="comet.common.pasteFailedInvalidFormat"
+                                defaultMessage="Paste failed, probably due to an invalid format"
+                            />
+                        ),
+                        error: e.toString(),
+                    });
+                    console.error("mutation failed", e);
+                }
+            }
+        } else {
+            console.error("Clidpboard is empty");
+        }
+    };
+
+    const handleCopyClick = async () => {
+        await writeClipboardText(JSON.stringify(await copyData!()));
+    };
+
+    return (
+        <>
+            <IconButton onClick={handleClick} size="large">
+                <MoreVertical />
+            </IconButton>
+            <Menu anchorEl={anchorEl} open={Boolean(anchorEl)} onClose={handleClose}>
+                {url && (
+                    <MenuItem
+                        key="copyUrl"
+                        onClick={() => {
+                            handleClose();
+                            writeClipboardText(url);
+                        }}
+                    >
+                        <ListItemIcon>
+                            <Domain />
+                        </ListItemIcon>
+                        <ListItemText primary={intl.formatMessage({ id: "comet.generic.copyUrl", defaultMessage: "Copy Url" })} />
+                    </MenuItem>
+                )}
+                {copyData && (
+                    <MenuItem
+                        onClick={async () => {
+                            setCopyLoading(true);
+                            await handleCopyClick();
+                            setCopyLoading(false);
+                            handleClose();
+                        }}
+                    >
+                        <ListItemIcon>{!copyLoading ? <Copy /> : <ThreeDotSaving />}</ListItemIcon>
+                        <ListItemText primary={intl.formatMessage({ id: "comet.generic.copy", defaultMessage: "Copy" })} />
```

Comment:
Please use messages from [messages.ts](https://github.com/vivid-planet/comet/blob/next/packages/admin/admin/src/messages.ts) here

Reply:
do we really have to use messages.ts in the library itself too?

Reply:
@dkarnutsch what do you think?

One problem when not using messages.ts could be that you're reusing the same ID (e.g. `comet.generic.copy`), but use different default messages (`Copy`, `Copy item`, `Copy this` etc.)

Reply:
Why don't you want to use the messages.ts?

For the sake of consistency I would always use messages.ts